content/docs/glossary/liquidity-pool.mdx

```diff
@@ -61,12 +63,256 @@ There are a few possibilities with regard to authorization. The behavior of the
+
+// kp1 issues the assets
+const kp1 = kps[0];
+const [ A, B ] = orderAssets(...[
+  new sdk.Asset("A", kp1.publicKey()),
+  new sdk.Asset("B", kp1.publicKey()),
+]);
+
+/// Establishes trustlines and pays out to `recipientKp` for all `assets`.
+function distributeAssets(issuerKp, recipientKp, ...assets) {
+  return server
+    .loadAccount(issuerKp.publicKey())
+    .then(issuer => {
+      const ops = assets.map(asset => [
+        sdk.Operation.changeTrust({
+          source: recipientKp.publicKey(),
+          limit: "100000",
+          asset: asset,
+        }),
+        sdk.Operation.payment({
+          source: issuerKp.publicKey(),
+          destination: recipientKp.publicKey(),
+          amount: "100000",
+          asset: asset,
+        }),
+      ]).flat();
+
+      let tx = buildTx(issuer, issuerKp, ...ops);
+      tx.sign(recipientKp);
+      return server.submitTransaction(tx);
+    });
+}
+
+function preamble() {
+  return Promise.all([1, 2].map(i => distributeAssets(kp1, kps[i], A, B)));
+}
+```
+
+</CodeExample>
+
+Here, we use `distributeAssets()` to establish trustlines and set up initial balances of two custom assets (`A` and `B`, issued by `kp1`) for two accounts (`kp2` and `kp3`).
+
+Note the `orderAssets()` helper here. Operations related to liquidity pools refer to the asset pair arbitrarily as `A` and `B`; however, they must be "ordered" such that `A < B`. This ordering is defined by the protocol, but its details should not be relevant (if you're curious, it's essentially lexographically ordered by asset type, code, then issuer). We can use the comparison methods built into the SDKs (like `Asset.compare`) to ensure we pass them in the right order and avoid errors.
+
+
+### Participation: Creation
+
+First, lets create a liquidity pool for the asset pair defined in the preamble. This involves establishing a trustline to the pool itself:
+
+<CodeExample>
+
+```js
+const poolShareAsset = new sdk.LiquidityPoolAsset(A, B, sdk.LiquidityPoolFeeV18);
+
+function establishPoolTrustline(account, keypair, poolAsset) {
+  return server.submitTransaction(
+    buildTx(account, keypair,
+      sdk.Operation.changeTrust({
+        asset: poolAsset,
+        limit: "100000"
+      })
+    )
+  );
+}
+```
+
+</CodeExample>
+
+This lets the participants hold pool shares (refer to the discussion about [pool shares earlier](#liquidity-pool-participation) for details), which means now they can perform deposits and withdrawals.
+
+### Participation: Deposits
+
+To work with a liquidity pool, you need to know its ID beforehand. It's a deterministic value, and only a single liquidity pool can exist for a particular asset pair, so you can calculate it locally from the pool parameters.
+
+<CodeExample>
+
+```js
+const poolId = sdk.getLiquidityPoolId(
+  "constant_product",
+  poolShareAsset.getLiquidityPoolParameters()
+).toString("hex");
+
+function addLiquidity(account, keypair, keypair, poolId, reserveA, reserveB) {
```

Comment:
Looks like there might be an extra `keypair` here?

Reply:
I think the naming convention from `buildTx` was a bit clearer, we could change the variables names like:
* account -> source
* keypair -> signer
* reserveA -> maxAmountA
* reserveB -> maxAmountB

It's clearer to name things with their intentions: "account" is too generic, while "sourceAccount" (or "source") is more specific to what it will be used for.

Reply:
Good points! I think making our examples in Typescript would increase clarity; that was my intent here with the naming.